### PR TITLE
Bring filters back into url, clear session storage on apply empty filter

### LINKED ISSUE #245: AMW_angular/#/deployments: filters come back
After a "Clear filter" and "Apply filter" go to a different page, if you come back to the deployment page, the filters are set again.
In this case, no filter should be set.

### LINKED ISSUE #250: AMW_angular/#/deployments: filter are no longer written to the url
Was this intentional?
It would be handy for users if they can just bookmark the site directly instead of: "Clipboard", past clipboard into address, load, bookmark.

### PR DESCRIPTION
**Pull request by RetGal**
Closes #245 &  closes #250

### CODE REVIEW

File: AMW_angular/io/src/app/deployment/deployments-list.component.ts [hunk 1]
```diff
@@ -176,7 +174,7 @@ export class DeploymentsListComponent {
   }
 
   logViewerLink(deploymentId: number){
-    window.location.href = '/AMW_web/pages/logView.xhtml?deploymentId=' + deploymentId + '&filters=' + encodeURI(JSON.stringify(this.filtersInUrl))
```
Comment (yvespp) [inline AMW_angular/io/src/app/deployment/deployments-list.component.ts]: I just added this, can't we keep it? See comment here: https://github.com/liimaorg/liima/pull/247#issuecomment-344621199

File: AMW_angular/io/src/app/deployment/deployments.component.ts [hunk 2]
```diff
@@ -665,4 +652,12 @@ export class DeploymentsComponent implements OnInit {
     this.isLoading = false;
   }
 
+  private goTo(destination: string) {
```
Comment (yvespp) [inline AMW_angular/io/src/app/deployment/deployments.component.ts]: goTo is a very generic name, can we call this updateFiltersInURL or something more descriptive?

File: AMW_angular/io/src/app/deployment/deployments.component.spec.ts [hunk 3]
```diff
@@ -332,7 +332,7 @@ describe('DeploymentsComponent (without query params)', () => {
       expect(deploymentsComponent.filters.length).toEqual(2);
   }));
 
-  it('should apply filters and add them to the sessionStorage on applyFilter',
+  it('should apply filters and add them to the sessionStorage on applyFilterss',
```
Comment (yvespp) [inline AMW_angular/io/src/app/deployment/deployments.component.spec.ts]: Typo

**Comment by RetGal** [comment]
@yvespp Done, I was not aware of  #247 (comment)

**Pull request merged**